pkg/gce-pd-csi-driver/node.go

```diff
@@ -95,13 +96,20 @@ func (ns *GCENodeServer) NodePublishVolume(ctx context.Context, req *csi.NodePub
 		return nil, status.Error(codes.Internal, fmt.Sprintf("cannot validate mount point: %s %v", targetPath, err))
 	}
 	if !notMnt {
-		// TODO(#95): check if mount is compatible. Return OK if it is, or appropriate error.
-		/*
-			1) Target Path MUST be the vol referenced by vol ID
-			2) TODO(#253): Check volume capability matches for ALREADY_EXISTS
-			3) Readonly MUST match
+		// Validate that the existing volume is compatible
+		partition := ""
+		if part, ok := req.GetVolumeContext()[common.VolumeAttributePartition]; ok {
+			partition = part
+		}
+		devicePath, err := ns.getDevicePath(volumeID, partition)
+		if err != nil {
+			return nil, status.Error(codes.Internal, fmt.Sprintf("Error when getting device path: %v", err))
+		}
+
+		if err := ns.volumeCompatible(stagingTargetPath, devicePath, volumeCapability); err != nil {
```

Comment:
This assumes that `targetPath` is bind mounted correctly to `devicePath`

Reply:
what do you mean by this? We are checking whether it is bind mounted correctly. If it is not it will throw an AlreadyExists error which is the expected behavior

Reply:
I mean that it is checking that `devicePath` (/dev/blah) is mounted to `stagingTargetPath` (NodeStage path), but not necessarily that `stagingTargetPath` is bind mounted to `targetPath` (NodePublish path).

Regardless, this is still a good check to have, it's just not actually verifying the Publish path.

Reply:
I see, I think in NodePublish we check that `/dev/sdx` is mounted to `publishTargetPath`. But you're right in that we never explicitly check that `stagingTargetPath` is bindmounted to `publishTargetPath`.

Edit: I just noted we check `stagingTargetPath` in `NodePublish` as well, it should actually be `targetPath` to make my above statement too.